app/controllers/public/sessions.js

```diff
@@ -27,15 +27,14 @@ export default class SessionsController extends Controller {
 
   @computed('model.dates', 'timezone')
   get allDates() {
-    let arr = [];
+    const arr = [];
     const sessions = groupBy(this.model.dates.toArray());
 
     Object.values(sessions).forEach(el => {
       arr.push(moment.tz(el[0].startsAt, this.timezone).toISOString());
     });
 
-    arr = arr.sort().filter((el, i) => moment(el).utc().format('YYYY-MM-DD') !== moment(arr[i + 1]).utc().format('YYYY-MM-DD')).map(el => el);
-    return arr;
+    return arr.sort().filter((el, i) => moment(el).utc().format('YYYY-MM-DD') !== moment(arr[i + 1]).utc().format('YYYY-MM-DD')).map(el => el);
```

Comment:
Instead of using filter like this, use Set or distinct/unique from lodash

Reply:
Sir Set only contain unique elements , and because every date is different , set will store all of them

Actually i am getting all the sessions and from the session i want to filter the date.

```js
dates = ["2021-09-17T03:40:00.000Z", "2021-09-17T04:50:00.000Z",  "2021-09-17T06:05:00.000Z", ​
 "2021-09-17T07:45:00.000Z"]
```

that's why what i am doing if the only the date part of the ISO string first element does not match with next element , then i return that date, 
instead if i use Set here it will store all the dates because every date is different

Reply:
Convert to date only and then use Set, obviously